src/main/java/org/broadinstitute/hellbender/engine/spark/datasources/ReadsSparkSink.java

```diff
@@ -156,40 +198,53 @@ private static void writeReadsADAM(
                 outputFile, Void.class, AlignmentRecord.class, AvroParquetOutputFormat.class, job.getConfiguration());
     }
 
-
-    private static void saveAsShardedHadoopFiles(JavaSparkContext ctx, String outputFile, JavaRDD<SAMRecord> reads, SAMFileHeader header, boolean writeHeader) throws IOException {
+    private static void saveAsShardedHadoopFiles(
+            final JavaSparkContext ctx, final String outputFile, final String referenceFile,
+            final SAMFormat samOutputFormat, final JavaRDD<SAMRecord> reads, final SAMFileHeader header,
+            final boolean writeHeader, final Configuration conf) throws IOException {
         // Set the static header on the driver thread.
-        SparkBAMOutputFormat.setHeader(header);
+        if (samOutputFormat == SAMFormat.CRAM) {
+            SparkCRAMOutputFormat.setHeader(header);
+        } else {
+            SparkBAMOutputFormat.setHeader(header);
+        }
 
         final Broadcast<SAMFileHeader> headerBroadcast = ctx.broadcast(header);
 
-        // SparkBAMOutputFormat is a static class, so we need to copy the header to each worker then call
-        final JavaRDD<SAMRecord> readsRDD = setHeaderForEachPartition(reads, headerBroadcast);
+        // SparkBAM/CRAMOutputFormat are a static classes, so we need to copy the header to each worker then call
+        final JavaRDD<SAMRecord> readsRDD = setHeaderForEachPartition(reads, samOutputFormat, headerBroadcast);
 
         // The expected format for writing is JavaPairRDD where the key is ignored and the value is SAMRecordWritable.
         final JavaPairRDD<SAMRecord, SAMRecordWritable> rddSamRecordWriteable = pairReadsWithSAMRecordWritables(headerBroadcast, readsRDD);
 
-        rddSamRecordWriteable.saveAsNewAPIHadoopFile(outputFile, SAMRecord.class, SAMRecordWritable.class, getOutputFormat(writeHeader));
+        rddSamRecordWriteable.saveAsNewAPIHadoopFile(outputFile, SAMRecord.class, SAMRecordWritable.class, getOutputFormat(samOutputFormat, writeHeader), conf);
```

Comment:
AFAIK, the "writable" part of SAMRecordWritable is used for record serialization during shuffle serialization, but the file output format is what dictates the file format (and record writer) for file save, so I think this gets the behavior we want.